File: vtstscripts-939/vef.pl
#!/usr/bin/env perl

use FindBin qw($Bin);

$plotflag = 1;
if(@ARGV > 0) { $plotflag = $ARGV[0]; }

$zip = $ENV{'VTST_ZIP'};
if($zip eq ''){ $zip = 'gzip'; }

$outzipped = 0;
if(-e "OUTCAR.gz") { $outzipped = 1; system "gunzip OUTCAR.gz"; }
if(-e "OUTCAR.bz2") { $outzipped = 1; system "bunzip2 OUTCAR.bz2"; }

$forces = `grep 'FORCES: max atom, RMS' OUTCAR`;
$energy = `grep 'energy  without entropy' OUTCAR`;

@forces = split /\n/, $forces;
@energy = split /\n/, $energy;

$num = @forces;

open OUT, ">fe.dat";

for($i=0; $i<$num; $i++){
    $line = $forces[$i] ;
    chomp($line) ;
    $line =~ s/^\s+//g;
    @line = split /\s+/,$line;
    $f = $line[4],"\n";

    $line = $energy[$i]; 
    chomp($line);
    $line =~ s/^\s+//g;
    @line = split /\s+/,$line;
    $e = $line[6],"\n";
    if(!$i) { $e0 = $e; }
    printf OUT "%5i %20.8f %20.6f %20.6g \n",$i,$f,$e,$e-$e0;
    printf "%5i %20.8f %20.6f %20.6g \n",$i,$f,$e,$e-$e0;
}

close OUT;

if($num>1 and $plotflag) {
    system "gnuplot $Bin/vef.gnu";
}

if($outzipped) { system "$zip -9 OUTCAR \&"; }
if($outzipped) { system "$zip -9 OUTCAR"; }

File: vtstscripts-939/vef.py
#!/usr/bin/env python
import aselite
from os.path import isfile, dirname, abspath, join
from os import system
from sys import exit, argv

if '-h' in argv:
    print 'usage: vef.py'
    print '       prints the force and energy for each ionic step of a vasp run'
    print
    exit(0)

vtst_path = dirname(abspath(__file__))

filename = 'OUTCAR'
if not isfile(filename):
    print 'No such file: %s' % filename
    exit(1)

traj = aselite.read_vasp_out(filename)
if len(traj) == 0:
    exit(0)

fe = open('fe.dat', 'w')
for i, atoms in enumerate(traj):
    e = atoms.get_potential_energy()
    if i == 0:
        e0 = e
    f = atoms.get_max_atom_force()
    str = '%5i %20.8f %20.6f %20.6g ' % (i,f,e,e-e0)
    print str
    fe.write(str+'\n')
fe.close()

if i > 1:
    system('gnuplot %s' % join(vtst_path, 'vef.gnu'))
